src/ai/llm/services/types.ts

```diff
@@ -9,9 +11,17 @@ export interface ILLMService {
     // Primary method for handling a user interaction from start to finish
     completeTask(userInput: string): Promise<string>;
 
-    // Update the system message/context
+    /**
+     * @deprecated Use setSystemPromptBuilder instead for modular system prompt support.
+     */
     updateSystemContext(newSystemPrompt: string): void;
 
+    // Accept a SystemPromptBuilder instance
+    setSystemPromptBuilder(builder: SystemPromptBuilder): void;
```

Comment:
i'm noticing we added this in both `ILLMService` and `MessageManager`. Is this intentional? seems like duplicate things and we now need to handle passing the objects around from LLMService to messagemanager